src/Directory.Packages.props

```diff
@@ -4,7 +4,8 @@
     <PackageVersion Include="BenchmarkDotNet" Version="0.13.5" />
     <PackageVersion Include="FluentAssertions" Version="6.11.0" />
     <PackageVersion Include="GitHubActionsTestLogger" Version="2.1.0" />
-    <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="1.0.0" />
+    <PackageVersion Include="Microsoft.Bcl.AsyncInterfaces" Version="6.0.0" />
+    <PackageVersion Include="Microsoft.Bcl.TimeProvider" Version="8.0.0-preview.4.23259.5" />
```

Comment:
While this is in preview, I think it best to target this PR into the `dotnet-8` branch and keep this out of main until it's stable.

Reply:
That's for like next 5 months or so right? I don't think there will be any changes in the code that work with `TimeProvider` so it might be fine.

Reply:
Yeah in November.

As-is though, it'll produce a warning from dotnet pack from depending on prerelease packages.

I think it's best to wait, rather than introduce prerelease packages into the dependency tree and suppress the warning.

Reply:
I have squashed the commits, do you think you can rebase the .net 8 branch on top of this?

I'll then try to fix some possible issues for .net 8 tests.

Reply:
Shall I pull this commit into my branch and push it up and then you'll fix-up the tests? Then it's "the right way around" with the .NET 8 preview 3 -> preview 4 -> TimeProvider sequencing.

Reply:
yup please, the build will fail temporarily, but I will work on fix.

Reply:
Cool - I'll do that now...

Reply:
Done - hopefully I haven't made too much of a mess of it 😅